src/root.zig

```diff
@@ -179,34 +195,40 @@ pub fn Algorithm(
                 .score = 0,
             };
 
-            const rows = needle.len;
-            const cols = haystack.len;
+            const haystack_normal = self.impl.convertString(haystack);
+            defer self.allocator.free(haystack_normal);
+
+            const needle_normal = self.impl.convertString(needle);
+            defer self.allocator.free(needle_normal);
```

Comment:
Maybe use `ArenaAllocator` for the lifetime of a single solve? Then implementations that don't need to allocate don't have to pay the price of a `dupe` to satisfy the free? Or else let the implementation manage the lifetimes?